### ISSUE TITLE
ITV:  Metadata change prevents download of high quality streams for recent programmes

### ISSUE THREAD

**Issue by stickenhoffen** [GetiPlayerAutomator/get-iplayer-automator#493]
Hi, I notice from source:

      NSArray *formatKeys = @[@"Flash - Very Low",@"Flash - Low",@"Flash - Standard",@"Flash - High"];
      NSArray *itvRateObjects = @[@"400",@"600",@"800",@"1200"];
      NSArray *bitrateObjects = @[@"400000",@"600000",@"800000",@"1200000"];

However, from XML returned:

        <MediaFile delivery="Streaming" bitrate="600000" base="" tkn="">
          <URL><![CDATA[mp4:production/priority/CATCHUP/1/0694/8663/001/1-0694-8663-001_PC01600_16X9.mp4]]></URL>
        </MediaFile>
        <MediaFile delivery="Streaming" bitrate="800000" base="" tkn="">
          <URL><![CDATA[mp4:production/priority/CATCHUP/1/0694/8663/001/1-0694-8663-001_PC01800_16X9.mp4]]></URL>
        </MediaFile>
        <MediaFile delivery="Streaming" bitrate="1200000" base="" tkn="">
          <URL><![CDATA[mp4:production/priority/CATCHUP/1/0694/8663/001/1-0694-8663-001_PC011200_16X9.mp4]]></URL>
        </MediaFile>
        <MediaFile delivery="Streaming" bitrate="1500000" base="" tkn="">
          <URL><![CDATA[mp4:production/priority/CATCHUP/1/0694/8663/001/1-0694-8663-001_PC011500_16X9.mp4]]></URL>
        </MediaFile>
        <MediaFile delivery="Streaming" bitrate="1800000" base="" tkn="">
          <URL><![CDATA[mp4:production/priority/CATCHUP/1/0694/8663/001/1-0694-8663-001_PC011800_16X9.mp4]]></URL>

Cheers.

**Comment by stickenhoffen** [comment]
I neglected to mention, streams which show the higher rates, do download but low quality.

Thanks.

**Comment by dinkypumpkin** [comment]
The metadata format changed slightly with the advent of higher-quality streams. That change breaks GiA's metadata parsing, so only the low-quality stream is picked up.   It looks like it started about a week ago, but older programmes still have metadata in the old format.  This must be fixed in code - there is no workaround.

**Comment by stickenhoffen** [comment]
Thanks for getting back, appreciate it.

**Comment by Vangelis66** [comment]
@stickenhoffen:  

Until the GiA code is patched to support the new higher ITV bitrates, 
you may wish to try the perl script get_flash_videos: 
http://code.google.com/p/get-flash-videos/
The maintainer of its ITV plugin has updated it 

v0.05:https://github.com/njtaylor/get-flash-videos/commit/3abd6e715062ed656705a2f052a1588a969efe02
v0.06:https://github.com/njtaylor/get-flash-videos/commit/fb64a705f892c14c75d6f77d9519a0c18366de98

to handle the new streams...
Use the latest combined script from: 
https://github.com/njtaylor/get-flash-videos/releases/
instead of the one linked to in the OSX installation guide...
(http://code.google.com/p/get-flash-videos/wiki/Installation#OSX)

Just tried it (on Win32) myself (from a UK IP address) 
on "Love Island 2015 - Se1Ep12" and 
```text
perl combined-get_flash_videos-1.24-git-20150615 -r 1800 "https://www.itv.com/itvplayer/love-island-2015/series-1/episode-12"
``` 
got me a 532 MB, 896x504 FLV file...

![1800_proof](https://cloud.githubusercontent.com/assets/9669492/8270045/46f28684-17d1-11e5-895c-6d909452c751.jpg)

--- Apologies for the off-topic nature of this comment; 
also, I am not responsible if you manage to break something 
in your system while trying GFV... ---

Regards.

**Comment by willson556** [comment]
Fixed in c5f25f4dbbb5f266bb96a3f8ac6c8249505227ae and higher quality downloads added in 1ac3b8d60460ca164acb157180c98bf297521413.
